### PR TITLE
Enable compression/chunking by default.

### PR DESCRIPTION
**Pull request by andre4i**
## Description

Compression used to be opt-in. This change enables compression by default and also changes the feature gate name in order to avoid collisions with the existing feature gates.

### CODE REVIEW

File: BREAKING.md [hunk 1]
```diff
@@ -48,6 +49,12 @@ The following interfaces available in `@fluidframework/runtime-definitions` are
 -   `IGarbageCollectionSnapshotData`
 -   `IGarbageCollectionSummaryDetailsLegacy`
 
+### Op compression is enabled by default
+
+If the size of a batch is larger than 614kb, the ops will be compressed. After upgrading to this version, if batches exceed the size threshold, the runtime will produce a new type of op with the compression properties. To be able to open a document which contains this type of op, the runtime version needs to be at least `client_v2.0.0-internal.2.3.0`. Older clients will close with assert `0x3ce` and will not be able to open the documents until they upgrade.
```
Comment (vladsud) [inline BREAKING.md:54]: I'd be a bit more specific here. I.e. app developer has to ensure that the version of the app with this version of FF can be deployed to prod when app developer ensured that 99.99% of app sessions are running with FF version above X.

**Review by vladsud — APPROVED** [review]

**Comment by vladsud** [comment]
Worth calling out - technically that's breaking change. I.e. user who jumped from 1.0 to 2.0.0-internal-4.0.0 a week ago would have trouble picking up this change. So theoretically we should put it into next.

**Comment by andre4i** [comment]
@vladsud, I agree in principle, however this is 2 major versions over the version which is incompatible. I think putting it in next would just add one extra version but nothing we can do release-wise can solve the issue of a deployment jumping major versions. Considering the feature kicks in only when there actually is a large payload (not that often) + the client will error out but only until it upgrades, I think this is very low risk..

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 2]
```diff
@@ -486,7 +486,15 @@ const defaultFlushMode = FlushMode.TurnBased;
 // We can't estimate it fully, as we
 // - do not know what properties relay service will add
 // - we do not stringify final op, thus we do not know how much escaping will be added.
-const defaultMaxBatchSizeInBytes = 950 * 1024;
+const defaultMaxBatchSizeInBytes = 700 * 1024;
+
+const defaultCompressionConfig = {
+	// Batches with content size exceeding this value will be compressed
+	minimumBatchSizeInBytes: 614400,
```
[the remark this answers is not in the document]
Comment (andre4i) [reply packages/runtime/container-runtime/src/containerRuntime.ts:493]: Not really. This only accounts for op content; compression doesn't observe the envelope size...

We ended up with this value after we experimented with internal customers. If we have very many ops, unfortunately compression will not help much, as we compress only the content, so the envelope overhead will be considerably larger and we'd hit that 1MB limit anyway.

**Pull request merged**